weave/trace/weave_client.py

```diff
@@ -1976,6 +1995,132 @@ def send_table_create() -> TableCreateRes:
 
         return table_ref
 
+    def _should_use_chunking(self, table: Table | WeaveTable) -> tuple[bool, bool]:
+        """Determine if we should use chunking and parallel chunks for a table."""
+        remote_request_bytes_limit = getattr(
+            self.server, "remote_request_bytes_limit", REMOTE_REQUEST_BYTES_LIMIT
+        )
+
+        # Primary heuristic: row count
+        use_chunking = len(table.rows) > ROW_COUNT_CHUNKING_THRESHOLD
+        # Secondary heuristic: basic size estimation for smaller tables
+        if not use_chunking and len(table.rows) > 0:
+            # Simple size estimation without full serialization
+            sample_row_size = len(str(table.rows[0]).encode("utf-8"))
+            estimated_bytes = sample_row_size * len(table.rows) * 2
+            use_chunking = estimated_bytes > remote_request_bytes_limit
+
+        # Determine parallel vs incremental chunking
+        use_parallel_chunks = False
+        if use_chunking:
+            test_req = TableCreateFromDigestsReq(
+                project_id=self._project_id(), row_digests=[]
+            )
+
+            def test_func(req: TableCreateFromDigestsReq) -> Any:
+                server = self.server
+                if hasattr(server, "_next_trace_server"):
+                    server = server._next_trace_server
+
+                assert hasattr(server, "_generic_request_executor")
+                assert hasattr(server._generic_request_executor, "__wrapped__")
+                return server._generic_request_executor.__wrapped__(
+                    server, "/table/create_from_digests", req
+                )
+
+            use_parallel_chunks = check_endpoint_exists(
+                test_func, test_req, "table_create_from_digests"
+            )
+
+        return use_chunking, use_parallel_chunks
+
+    def _create_table_with_parallel_chunks(
+        self, table: Table | WeaveTable
```

Comment:
why is this `Table | WeaveTable`?

Reply:
follows the existing implementation of _save_table